en/views/json-and-xml-views.rst

```diff
@@ -216,15 +216,15 @@ mappings in your controller::
             // Set Out Format View
             $this->viewBuilder()->className($formats[$format]);
 
-            // Set Force Download
-            $this->response->download('report-' . date('YmdHis') . '.' . $format);
-
             // Get data
             $videos = $this->Videos->find('latest');
 
             // Set Data View
             $this->set(compact('videos'));
             $this->set('_serialize', ['videos']);
+
+            // Set Force Download
+            return $this->response->download('report-' . date('YmdHis') . '.' . $format);
```

Comment:
Wouldn't in this case ``withDownload()`` be more appropriate, as it is not deprecated?

https://api.cakephp.org/3.5/class-Cake.Http.Response.html#_download

Reply:
Perfectly good catch!